src/patchelf.cc

```diff
@@ -1885,9 +1906,186 @@ void ElfFile<ElfFileParamNames>::clearSymbolVersions(const std::set<std::string>
         }
     }
     changed = true;
-    this->rewriteSections();
+    rewriteSections();
 }
 
+struct symstr {
```

Comment:
Perhaps one could use a std::string_view for this?

Reply:
maybe... This looks most like the original C-implementation I did. However, I need none of the 'burden' of string_view.

Reply:
looked into string_view and the constructor string_view(char *s) can be useful to replace the argz_* functions too. Good tip,